pennylane/gradients/jvp.py

```diff
@@ -304,3 +322,12 @@ def _single_measurement_zero(m, tangent):
     res = qml.math.convert_like(np.zeros(dim), tangent)
     res = qml.math.cast_like(res, tangent)
     return res
+
+
+def _shots_copies(shot_vector):
```

Comment:
2 points:
* This function definitely feels more general than to be kept here - how about moving it into `_device.py`?
* I think the name could be made more descriptive, how about `_get_num_copies`?

Reply:
Done, thanks for the good suggestions!